modules/picarto.js

```diff
@@ -107,27 +107,80 @@ function getChannel(stream) {
 	return stream
 }
 
-function getToken(stream) {
+function getToken(stream, cookieJar) {
+	var cookieJar = cookieJar || request.jar();
 	return new Promise(function (resolve, reject) {
-		jsdom.env({
-			url: "https://picarto.tv/" + getChannel(stream),
-			features: {
-				FetchExternalResources: ["script"],
-				ProcessExternalResources: ["script"]
-			},
-			done: function (error, window) {
-				if (error) { console.log(error); reject("jsdom error"); }
-				try {
-					var $ = window.$;
-					resolve({
-						token: getTokenFromHTML($("body").html())
-					});
-				} catch (e) {
-					reject("channelDoesNotExist");
-				}
+		request({
+			uri: "https://picarto.tv/" + getChannel(stream),
+			jar: cookieJar,
+			headers: {'Referer': 'https://www.picarto.tv/'}
+		}, function (error, response, body) {
+
+			if (error) {
+				reject(error);
+				return;
+			}
+
+			if (!response || (response.statusCode !== 200 && response.statusCode !== 302)) {
+				reject(new Error('Unexpected status code while fetching auth token: ' + response.statusCode));
+				return;
+			}
+
+			var authToken = getTokenFromHTML(body);
+			if (!authToken) {
+				reject( new Error('Error parsing auth token'));
+			} else {
+				resolve(authToken);
 			}
 		});
-	})
+	});
+}
+
+function getAuthedCookieJar(username, password) {
+	var cookieJar = request.jar();
+	return new Promise(function (resolve, reject) {
+		console.log("getAuthedCookieJar: ", username, password);
```

Comment:
I am sure you did test it! Doesn't mean that I won't test it myself again :wink:. But yeah, it's the only debug command I found, so removing that should make it OK